homeassistant/components/sleepiq/__init__.py

```diff
@@ -99,6 +99,13 @@ def name(self):
             self.bed.name, self.side.sleeper.first_name, self._name
         )
 
+    @property
+    def unique_id(self):
+        """Return a unique ID for the bed."""
+        return "{}-{}-{}".format(
+            self._bed_id, self._side, self._name
```

Comment:
This does not satisfy our requirements for unique IDs, for more information see:
<https://developers.home-assistant.io/docs/entity_registry_index/#unique-id-requirements>

Reply:
Hm. Best as I can tell from snooping a web browser session with my sleepiq account, _bed_id is a unique identifier, though I can't confirm if it's a serial number. My bed's pump has a MAC address listed on it, but I'm not seeing it exported in the sleepiq API anywhere (nor am I seeing it pop up in any traffic sniffing poking around sleepiq.sleepnumber.com for my bed). So I think this satisfies the "unique and not possible for the user to change" requirement.

From snooping web traffic:
```
{beds: [{status: 1, bedId: "-9223372020241268527",…}]}
 beds: [{status: 1, bedId: "-9223372020241268527",…}]
    0: {status: 1, bedId: "-9223372020241268527",…}
      bedId: "-9223372020241268527"
      leftSide: {isInBed: false, alertDetailedMessage: "Bed needs to be baselined", sleepNumber: 65, alertId: 2,…}
      rightSide: {isInBed: false, alertDetailedMessage: "Bed needs to be baselined", sleepNumber: 65, alertId: 2,…}
      status: 1
```
From poking at the bed with sleepyq python lib:
```
>>> bed_id = client.default_bed_id(None)
>>> print(bed_id)
-9223372020241268527
```

My bed is then exporting four different sensor, two for each side of the bed, one a binary_sensor is_in_bed and one a numerical sensor for sleepnumber. So the created unique_ids here should be "-9223372020241268527-left-sleepnumber", "-9223372020241268527-right-sleepnumber", "-9223372020241268527-left-is_in_bed" and "-9223372020241268527-right-is_in_bed".

And apparently, I need to baseline my bed, even though I've definitely done that already...

Reply:
Argh, my comment got eaten. Haven't used github much for anything but push/pull anytime lately. But basically, I've hit "Resolve conversation" in hopes that it's the right thing to do for a re-evaluation of the satisfaction of unique ID rules, based on my more detailed explanation above.

Reply:
They do, yes, but I'm not quite sure what the problem is with using the unique bed ID provided by the sleepiq API is with respect to satisfying the HA uniqueness requirements.

Reply:
> but I'm not quite sure what the problem is with using the unique bed ID provided 

I'm not sure either, however, the name component of it definitely is not acceptable for our requirements.

Reply:
Ah, I see, I didn't understand which part of the scheme didn't look valid. The _name bit is actually hard-coded in the integration, it's either "is_in_bed" or "sleep_number", from homeassistant/components/sleepiq/const.py, which maps from "isInBed" and "sleepNumber" fields returned in the API -- see screenshot below.

![image](https://user-images.githubusercontent.com/506884/152919181-41090472-65f9-436b-87a7-4582cb38b136.png)

Reply:
Perhaps the integration should have that _name field renamed to _sensor_type or something like that to clarify?

Reply:
Aah check! That solves the confusion 👍 

Yeah, or maybe add a comment here @ the unique ID to clarify that.

Reply:
Also, turns out I was doing this slightly wrong. The _name field was actually not the right thing to be using after all, as that's a friendly output string version, what we actually wanted was self.type there, but it wasn't being filled in by the binary sensor, which is now corrected w/in this MR as well.